nipoppy/workflows/dataset_init.py

```diff
@@ -85,13 +107,65 @@ def run_main(self):
             FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest, log_level=logging.DEBUG
         )
 
+        if self.bids_source is not None:
+            self._init_manifest_from_bids_dataset()
+
         # inform user to edit the sample files
         self.logger.warning(
             f"Sample config and manifest files copied to {self.layout.fpath_config}"
             f" and {self.layout.fpath_manifest} respectively. They should be edited"
             " to match your dataset"
         )
 
+    def _init_manifest_from_bids_dataset(self) -> None:
+        """Assume a BIDS dataset with session level folders.
+
+        No BIDS validation is done.
+        """
+        df = {"participant_id": [], "visit_id": [], "session_id": [], "datatype": []}
```

Comment:
With `from nipoppy.tabular.manifest import Manifest`:
```suggestion
        df = {
            Manifest.col_participant_id: [],
            Manifest.col_visit_id: [],
            Manifest.col_session_id: [],
            Manifest.col_datatype: [],
        }
```

to avoid hardcoding of column names

Reply:
Also I think it is a little confusing to name this `df` since it is not yet a dataframe